### PR TITLE
Disable verify transitive dependencies check with manual integration

### LINKED ISSUE #5851: Should verify_no_static_framework_transitive_dependencies execute if integrate_targets is false?
I don't see a reason to execute this check if automatic integration is turned off and leave it up to the developer to control which dependencies get linked and where.

### PR DESCRIPTION
**Pull request by dnkoutso**
closes #5851

### CODE REVIEW

File: lib/cocoapods/installer.rb [hunk 1]
```diff
@@ -110,6 +110,11 @@ def install!
       resolve_dependencies
       download_dependencies
       verify_no_duplicate_framework_and_library_names
+      if installation_options.integrate_targets?
+        verify_no_static_framework_transitive_dependencies
+      else
+        UI.section 'Skipping Verifying Framework Transitive Dependencies Check'
+      end
       verify_no_static_framework_transitive_dependencies
```
Comment (dnkoutso) [inline lib/cocoapods/installer.rb]: oops

**Comment by orta** [comment]
Failure seems legit:

```
Bacon::Error: #<Bacon::Context:0x007fc196259a68 @name="#verify_no_static_framework_transitive_dependencies", @after=[], @before=[#<Proc:0x007fc194d64e50@spec/unit/installer_spec.rb:50>, #<Proc:0x007fc196259310@spec/unit/installer_spec.rb:340>], @block=#<Proc:0x007fc196259ba8@spec/unit/installer_spec.rb:339>, @podfile=#<Pod::Podfile:0x007fc1961fad10 @defined_in_file=nil, @internal_hash={"installation_method"=>{"name"=>"cocoapods", "options"=>{"integrate_targets"=>false}}}, @root_target_definitions=[#<Pod::Podfile::TargetDefinition label=Pods>], @current_target_definition=#<Pod::Podfile::TargetDefinition label=Pods>>, @installer=#<Pod::Installer:0x007fc1961e8368 @sandbox=#<Pod::Sandbox> with root /Users/travis/build/CocoaPods/CocoaPods/tmp/Pods, @podfile=#<Pod::Podfile:0x007fc1961fad10 @defined_in_file=nil, @internal_hash={"installation_method"=>{"name"=>"cocoapods", "options"=>{"integrate_targets"=>false}}}, @root_target_definitions=[#<Pod::Podfile::TargetDefinition label=Pods>], @current_target_definition=#<Pod::Podfile::TargetDefinition label=Pods>>, @lockfile=#<Pod::Lockfile>, @use_default_plugins=true>, @file=#<Pathname:/yolo.m>, @lockfile=#<Pod::Lockfile>, @lib_thing=#<Pathname:/libThing.a>>.raise?(Pod::Informative) failed
	spec/unit/installer_spec.rb:366:in `block (3 levels) in <module:Pod>': #verify_no_static_framework_transitive_dependencies - detects transitive static dependencies which are linked directly to the user target

```

**Comment by dnkoutso** [comment]
Should be OK now, by default the `intergate_targets` config option was set to `false`.

File: spec/unit/installer_spec.rb [hunk 2]
```diff
@@ -147,6 +147,18 @@ def @installer.analyze(*)
...
+      it 'performs framework transitive dependencies check if corresponding config is set' do
+        @installer.installation_options.integrate_targets = true
+        @installer.expects(:verify_no_static_framework_transitive_dependencies)
+        @installer.install!
+      end
+
+      it 'skips framework transitive dependencies check if corresponding config is not set' do
+        @installer.installation_options.integrate_targets = false
+        @installer.expects(:verify_no_static_framework_transitive_dependencies).never
+        @installer.install!
```
Comment (segiddins) [inline spec/unit/installer_spec.rb]: please also verify the warning is printed

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,7 +8,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-* None.  
+* Disable verify transitive dependencies check with manual integration.  
+   [Dimitris Koutsogiorgas](https://github.com/dnkoutso)
```
Comment (segiddins) [inline CHANGELOG.md]: there are extra leading spaces on this line and the next

Comment (dnkoutso) [inline CHANGELOG.md]: Fixed

File: spec/unit/installer_spec.rb (hunk 2 above)
Comment (dnkoutso) [inline spec/unit/installer_spec.rb]: DoneDone

**Comment by dnkoutso** [comment]
Rebased. Lets 🙏 for green.

**Comment by dnkoutso** [comment]
Will close that for now until I figure out test failures.

**Pull request closed without merging**